server/plugins/schema-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/schema/obmysql/parser/OBMySQLGetDBTableByParser.java

```diff
@@ -244,33 +246,102 @@ public DBTablePartition getPartition() {
         if (partitionStmt.getSubPartitionOption() == null) {
             return partition;
         }
-        // TODO 目前 ODC 仅支持 HASH/KEY 二级分区, 其它类型后续需补充
         partition.setSubpartitionTemplated(partitionStmt.getSubPartitionOption().getTemplates() != null);
-        SubPartitionOption subOption = partitionStmt.getSubPartitionOption();
         String type = partitionStmt.getSubPartitionOption().getType();
         if ("key".equals(type.toLowerCase())) {
-            subPartitionOption.setType(DBTablePartitionType.KEY);
-            subPartitionOption.setColumnNames(subOption.getSubPartitionTargets() == null ? null
-                    : subOption.getSubPartitionTargets().stream().map(item -> removeIdentifiers(item.getText()))
-                            .collect(Collectors.toList()));
-            subPartitionOption
-                    .setPartitionsNum(partition.getSubpartitionTemplated() ? subOption.getTemplates().size()
-                            : partitionStmt.getPartitionElements().get(0).getSubPartitionElements().size());
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.KEY,
+                    false);
         } else if ("hash".equals(type.toLowerCase())) {
-            subPartitionOption.setType(DBTablePartitionType.HASH);
-            Expression expression = subOption.getSubPartitionTargets().get(0);
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.HASH,
+                    true);
+        } else if ("range".equals(type.toLowerCase())) {
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.RANGE,
+                    true);
+        } else if ("range columns".equals(type.toLowerCase())) {
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt,
+                    DBTablePartitionType.RANGE_COLUMNS, false);
+        } else if ("list".equals(type.toLowerCase())) {
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt, DBTablePartitionType.LIST,
+                    true);
+        } else if ("list columns".equals(type.toLowerCase())) {
+            parseSubPartition(subPartitionOption, subPartitionDefinitions, partitionStmt,
+                    DBTablePartitionType.LIST_COLUMNS, false);
+        } else {
+            partition.setWarning("Only support HASH/KEY subpartition currently");
+        }
+        return partition;
+    }
+
+    private void parseSubPartition(DBTablePartitionOption subPartitionOption,
```

Comment:
Actually, this method is not related to `parse` itself. `FillSubPartition` may be more clear.

Reply:
ok